ion/cmp_test.go

```diff
@@ -0,0 +1,145 @@
+package ion
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+type ionEqual interface {
+	eq(other ionEqual) bool
+}
+
+type ionFloat struct{ float64 }
+type ionDecimal struct{ *Decimal }
+type ionTimestamp struct{ time.Time }
+
+func (thisFloat ionFloat) eq(other ionEqual) bool {
+	return cmp.Equal(thisFloat.float64, other.(ionFloat).float64, cmpopts.EquateNaNs())
+}
+
+func (thisDecimal ionDecimal) eq(other ionEqual) bool {
+	if val, ok := other.(ionDecimal); ok {
+		if thisDecimal.scale != val.scale {
+			return false
+		}
+		return thisDecimal.Decimal.Equal(val.Decimal)
+	}
+	return false
+}
+
+func (thisTimestamp ionTimestamp) eq(other ionEqual) bool {
+	if val, ok := other.(ionTimestamp); ok {
+		return thisTimestamp.Time.Equal(val.Time)
+	}
+	return false
+}
+
+func cmpAnnotations(thisAnnotations, otherAnnotations []string) bool {
+	if len(thisAnnotations) != len(otherAnnotations) {
+		return false
+	}
+
+	for idx, this := range thisAnnotations {
+		other := otherAnnotations[idx]
+
+		if !cmp.Equal(this, other) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func cmpValueSlices(thisValues, otherValues []interface{}) bool {
+	if len(thisValues) == 0 && len(otherValues) == 0 {
+		return true
+	}
+
+	if len(thisValues) != len(otherValues) {
+		return false
+	}
+
+	res := false
+	for idx, this := range thisValues {
+		other := otherValues[idx]
+		thisType, otherType := reflect.TypeOf(this), reflect.TypeOf(other)
+
+		if thisType != otherType {
+			return false
+		}
+
+		switch this.(type) {
+		case string: // null.Sexp, null.List, null.Struct
```

Comment:
Nit: the typed null syntax is always lowercase (`null.sexp`, `null.list`, etc)

Reply:
Thanks